src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React from 'react';
+import {
+  actions,
+  ActionType,
+  ColumnInstance,
+  HeaderGroup,
+  Hooks,
+  makePropGetter,
+  TableInstance,
+  TableKeyedProps,
+  TableState,
+  useGetLatest,
+} from 'react-table';
+
+export const useResizeColumns = <T extends Record<string, unknown>>(
+  ownerDocument: Document | undefined,
+) => (hooks: Hooks<T>) => {
+  hooks.getResizerProps = [defaultGetResizerProps(ownerDocument)];
+  hooks.stateReducers.push(reducer);
+  hooks.useInstanceBeforeDimensions.push(useInstanceBeforeDimensions);
+};
+
+const defaultGetResizerProps = (ownerDocument: Document | undefined) => (
+  props: TableKeyedProps,
+  {
+    instance,
+    header,
+    nextHeader,
+  }: {
+    instance: TableInstance;
+    header: HeaderGroup;
+    nextHeader: HeaderGroup;
+  },
+) => {
+  if (!ownerDocument) {
+    return props;
+  }
+
+  const { dispatch, flatHeaders } = instance;
+
+  const onResizeStart = (
+    e: React.TouchEvent & React.MouseEvent,
+    header: HeaderGroup,
+  ) => {
+    let isTouchEvent = false;
+    if (e.type === 'touchstart') {
+      // lets not respond to multiple touches (e.g. 2 or 3 fingers)
+      if (e.touches && e.touches.length > 1) {
+        return;
+      }
+      isTouchEvent = true;
+    }
+
+    // Setting `width` here because it might take several rerenders until actual column width is set.
+    flatHeaders.forEach((h) => {
+      if (!h.width) {
+        h.width = h.resizeWidth;
+      }
+    });
+
+    const headerIdWidths = getLeafHeaders(header).map((d) => [d.id, d.width]);
+    const nextHeaderIdWidths = nextHeader
+      ? getLeafHeaders(nextHeader).map((d) => [d.id, d.width])
+      : [];
+
+    const clientX = isTouchEvent ? Math.round(e.touches[0].clientX) : e.clientX;
+
+    const dispatchMove = (clientXPos: number) =>
+      dispatch({ type: actions.columnResizing, clientX: clientXPos });
+    const dispatchEnd = () =>
+      dispatch({
+        type: actions.columnDoneResizing,
+      });
+
+    const handlersAndEvents = {
+      mouse: {
+        moveEvent: 'mousemove',
+        moveHandler: (e: MouseEvent) => dispatchMove(e.clientX),
+        upEvent: 'mouseup',
+        upHandler: () => {
+          ownerDocument.removeEventListener(
+            'mousemove',
+            handlersAndEvents.mouse.moveHandler,
+          );
+          ownerDocument.removeEventListener(
+            'mouseup',
+            handlersAndEvents.mouse.upHandler,
+          );
+          dispatchEnd();
+        },
+      },
+      touch: {
+        moveEvent: 'touchmove',
+        moveHandler: (e: TouchEvent) => {
+          if (e.cancelable) {
+            e.preventDefault();
+            e.stopPropagation();
+          }
+          dispatchMove(e.touches[0].clientX);
+          return false;
```

Comment:
Sure you need return boolean? mouse move handler is void, so types are different

Reply:
Removed